tests/test_qnode.py

```diff
@@ -492,6 +492,26 @@ def circuit(x):
 
         assert np.allclose(grad, 0)
 
+    def test_unrecognized_kwargs_raise_warning(self):
+        """Test that passing gradient_kwargs not included in qml.gradients.SUPPORTED_GRADIENT_KWARGS raises warning"""
+        dev = qml.device("default.qubit", wires=2)
+
+        with warnings.catch_warnings(record=True) as w:
+
+            @qml.qnode(dev, grad_method=qml.gradients.finite_diff)
+            def circuit(params):
+                qml.RX(params[0], wires=0)
+                return qml.expval(qml.PauliZ(0)), qml.var(qml.PauliZ(0))
+
+            @qml.qnode(dev, random_kwarg=qml.gradients.finite_diff)
```

Comment:
Since we are needing to test for two separate logical flows, one got 'gradient_fn' and one for random keyword arguments, we could just split the test in two. That way if something fails, it will be easier to localize which logical path is failing.

Not that important, but if tests can be broken up easily without having to do a bunch of duplicated set up, they usually should be broken up. Nothing wrong with precise, tiny tests.